vars/dockerExecute.groovy

```diff
@@ -161,43 +169,24 @@ void call(Map parameters = [:], body) {
                         body()
                     }
                 } else {
-                    if(!config.dockerName){
-                        config.dockerName = UUID.randomUUID().toString()
-                    }
-
-                    Map paramMap = [
+                    dockerExecuteOnKubernetes(
                         script: script,
-                        containerCommands: [:],
-                        containerEnvVars: [:],
-                        containerPullImageFlags: [:],
-                        containerMap: [:],
                         containerName: config.dockerName,
-                        containerPortMappings: [:],
-                        containerWorkspaces: [:],
-                        stashContent: config.stashContent
-                    ]
-
-                    paramMap.containerCommands[config.sidecarImage] = ''
-
-                    paramMap.containerEnvVars[config.dockerImage] = config.dockerEnvVars
-                    paramMap.containerEnvVars[config.sidecarImage] = config.sidecarEnvVars
-
-                    paramMap.containerPullImageFlags[config.dockerImage] = config.dockerPullImage
-                    paramMap.containerPullImageFlags[config.sidecarImage] = config.sidecarPullImage
-
-                    paramMap.containerMap[config.dockerImage] = config.dockerName
-                    paramMap.containerMap[config.sidecarImage] = config.sidecarName
-
-                    paramMap.containerPortMappings = config.containerPortMappings
-
-                    paramMap.containerWorkspaces[config.dockerImage] = config.dockerWorkspace
-                    paramMap.containerWorkspaces[config.sidecarImage] = ''
-
-                    dockerExecuteOnKubernetes(paramMap){
-                        echo "[INFO][${STEP_NAME}] Executing inside a Kubernetes Pod with sidecar container"
-                        if(config.sidecarReadyCommand) {
```

Comment:
[clarification] waiting for the side car container has been moved inside `dockerExecuteOnKubernetes`